packages/@aws-cdk/aws-cognito/lib/user-pool-client.ts

```diff
@@ -461,6 +461,16 @@ export class UserPoolClient extends Resource implements IUserPoolClient {
             },
             physicalResourceId: PhysicalResourceId.of(this.userPoolClientId),
           },
+          onUpdate: {
+            region: Stack.of(this).region,
```

Comment:
Is there any circumstance where the custom resources wouldn't be in the same region? Probably a dumb question on my end, but I just want to check.

Reply:
@TheRealAmazonKendra That's a great question and honestly I am not entirely sure... So the User Pool and the Client always have to be in the same region. There also isn't (currently) a `fromXXX` static method on `UserPoolClient` that lets you specify region (there's `fromUserPoolClientId` but not `fromUserPoolClientArn`). And while User Pool has a `fromUserPoolArn` and `IUserPool` has `addClient`, I actually feel like that would fail in CloudFormation since the pool is in another region.

And then `fromUserPoolClientId` actually has an implementation of `get userPoolClientSecret()` that always throws (probably because it doesn't take an `IUserPool` as a parameter to make the API call).

I am not sure that there's any situation where the `region` parameter's value could be different from the region where the pool/client exist (and have things actually work with the interfaces that exist today).

So I guess `region` could actually probably be omitted? But frankly, I just copied what had been done for `onCreate`.

Which, I think, means the options are to either:
1. Keep `region` as-is
2. Remove `region` from `onCreate` and `onUpdate`
3. Change both to `this.userPool.env.region` (which at the moment should always be the stack's region)